src/coreclr/jit/patchpoint.cpp

```diff
@@ -47,12 +47,6 @@ class PatchpointTransformer
     //   Number of patchpoints transformed.
     int Run()
     {
-        // If the first block is a patchpoint, insert a scratch block.
-        if (compiler->fgFirstBB->HasFlag(BBF_PATCHPOINT))
-        {
-            compiler->fgEnsureFirstBBisScratch();
-        }
-
```

Comment:
You should keep this or its moral equivalent, Tier0 instr needs to insert some execute once code at method entry

Reply:
I would've thought the init BB would be sufficient. It should have been created before this.

Reply:
Perhaps the issue is that the `bbNum` "is backedge" approximation is placing patchpoints in the init BB. Seems like we can teach the logic to never place a patchpoint in the init BB (or, more ideally, switch to a DFS tree based approach – but that's maybe too expensive for tier0).

Reply:
The importer places the patchpoint in the init BB because of stress.

What is the point of the assert? Is it simply used as an attempt to verify "is not part of a cycle"? Seems like the assert should just be removed if so, given the stress mode.

Reply:
I ended up removing the assert and also changing code below to make sure to insert the initialization of the patchpoint at the beginning of the init BB.